packages/backend/src/utils/catalogUtils.ts

```diff
@@ -26,7 +26,19 @@ export enum CatalogFormat {
   UNKNOWN = 'unknown',
 }
 
-export function sanitize(raw: object & { version: string }): ApplicationCatalog {
+export function sanitize(rawObject: object): ApplicationCatalog {
+  // if there is no version in the user catalog, we try to adapt it automatically to the CURRENT format
+  let raw: object & { version: string };
+  if (!('version' in rawObject)) {
+    raw = {
+      ...rawObject,
+      version: CatalogFormat.CURRENT,
```

Comment:
Setting any unknown file version to current here (before even trying to adapt) threw me off at first. IMHO it should be `adaptToCurrent(raw:object): object & { version: string }` which adds the version as the last step before returning. (and maybe throws an error if the file is missing some key element and can't be converted)